app/controllers/application_controller.rb

```diff
@@ -1,6 +1,41 @@
 class ApplicationController < ActionController::API
 
   private
+  set_current_tenant_through_filter
+  before_action :set_the_current_tenant
+
+  def set_the_current_tenant
+    return unless ENV["ENFORCE_TENANCY"]
+    return unless ident = user_identity
+
+    tenant = Tenant.find_by(:external_tenant => ident)
+    if tenant
+      set_current_tenant(tenant)
+    else
+      render :json => { :errors => "Unauthorized" }, :status => :unauthorized
```

Comment:
The error document should probably be a little bit more descriptive.  See https://jsonapi.org/format/#errors and https://github.com/ManageIQ/topological_inventory-api/pull/109/files#diff-df731b83f826451eea12d418a5da8d51R8 for an example